### PR TITLE
feat: add client to deployment

### PR DESCRIPTION
**Pull request by alaeddine-13**
based on: https://github.com/jina-ai/jina/pull/5574
closes: https://github.com/jina-ai/jina/issues/5551

### CODE REVIEW

File: docs/concepts/executor/serve.md [hunk 1]
```diff
@@ -59,7 +58,8 @@ class MyExec(Executor):
 
 
 with Deployment(uses=MyExec, port=12345, replicas=2) as dep:
-    dep.block()
+    docs = dep.post(inputs=DocumentArray.empty(1), on='/foo')
```
Comment (JoanFM) [inline docs/concepts/executor/serve.md:61]: put on in front, which is more aligned with the typical use of Flow

File: docs/concepts/executor/serve.md [hunk 2]
```diff
@@ -59,7 +58,8 @@ class MyExec(Executor):
...
+    print(docs.texts)
 ```
```
Comment (JoanFM) [inline docs/concepts/executor/serve.md:63]: example of output?

**Review by JoanFM — DISMISSED** [review]

File: docs/concepts/executor/serve.md (hunk 2 above)
Comment (alaeddine-13) [reply docs/concepts/executor/serve.md:63]: image
already there for each type
https://feat-add-client-deployment--jina-docs.netlify.app/concepts/executor/serve/#serve-directly

File: docs/concepts/executor/serve.md (hunk 1 above)
Comment (alaeddine-13) [inline docs/concepts/executor/serve.md:61]:
```suggestion
    docs = dep.post(on='/foo', inputs=DocumentArray.empty(1))
```

File: docs/concepts/executor/serve.md [hunk 3]
```diff
@@ -75,7 +75,8 @@ py_modules:
 from jina import Deployment
 
 with Deployment(uses='executor.yaml', port=12345, replicas=2) as dep:
-    dep.block()
+    docs = dep.post(inputs=DocumentArray.empty(1), on='/foo')
```
Comment (alaeddine-13) [inline docs/concepts/executor/serve.md:78]:
```suggestion
    docs = dep.post(on='/foo', inputs=DocumentArray.empty(1))
```

File: docs/concepts/executor/serve.md [hunk 4]
```diff
@@ -85,7 +86,8 @@ with Deployment(uses='executor.yaml', port=12345, replicas=2) as dep:
 from jina import Deployment
 
 with Deployment(uses='jinaai://my-username/MyExec/', port=12345, replicas=2) as dep:
-    dep.block()
+    docs = dep.post(inputs=DocumentArray.empty(1), on='/foo')
```
Comment (alaeddine-13) [inline docs/concepts/executor/serve.md:89]:
```suggestion
    docs = dep.post(on='/foo', inputs=DocumentArray.empty(1))
```

File: docs/concepts/executor/serve.md [hunk 5]
```diff
@@ -96,7 +98,8 @@ with Deployment(uses='jinaai://my-username/MyExec/', port=12345, replicas=2) as
 from jina import Deployment
 
 with Deployment(uses='docker://my-executor-image', port=12345, replicas=2) as dep:
-    dep.block()
+    docs = dep.post(inputs=DocumentArray.empty(1), on='/foo')
```
Comment (alaeddine-13) [inline docs/concepts/executor/serve.md:101]:
```suggestion
    docs = dep.post(on='/foo', inputs=DocumentArray.empty(1))
```

**Review by JoanFM — APPROVED** [review]

**Pull request merged**